src/182B30.c

```diff
@@ -215,7 +215,51 @@ void func_80257B88(void) {
     func_802571F0(1, gBattleStatus.partnerActor);
 }
 
-INCLUDE_ASM(s32, "182B30", update_player_actor_shadow);
+void update_player_actor_shadow(void) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Camera* camera = &gCameras[CAM_BATTLE];
+    Actor* player = battleStatus->playerActor;
+    ActorPart* parts = player->partsTable;
+    Shadow* shadow;
+    f32 x, y, z, scale;
+
+    parts->unk_8C = spr_update_player_sprite(0, parts->currentAnimation, parts->animationRate);
+
+    if (player->flags & ACTOR_FLAG_10000000) {
+        func_802549F4(player);
+    }
+
+    shadow = get_shadow_by_index(player->shadow);
+    shadow->flags &= ~SHADOW_FLAGS_1;
+
+    if (!battleStatus->outtaSightActive) {
+        shadow->unk_05 = 128;
+    } else {
+        shadow->unk_05 = 40;
+    }
+
+    scale = 32767.0f;
```

Comment:
As an arg to npc_raycast_down_sides this is a distance (on input, max raycast distance, on output, the distance the cast went). Scale works since it gets given to set_standard_shadow_scale but I think I'd prefer `distance` still (especially because it seems like `set_standard_shadow_scale` takes a distance-from-floor, not a scale, and converts it into a scale factor of inverse proportion?)